pkg/network/errors/errors.go

```diff
@@ -0,0 +1,13 @@
+package errors
```

Comment:
To remove redundancy we can just move `errors.go` to `pkg/network/errors.go` since there is no sub package at `pkg/network` using it and rename it to `CriticalError` so we will have `network.CriticalError`.

Or we can even do a type alias to export unexported `networkCriticalError` as `network.CriticalError`.

But this is just cosmetics.